schedule/staging/ext4@s390x-staging.yaml

```diff
@@ -34,6 +35,11 @@ schedule:
   - console/validate_blockdevices
   - console/validate_free_space
   - console/validate_read_write
+conditional_schedule:
+  validate_beta_popup:
+    BETA:
+      1:
```

Comment:
Maybe it would make it less complicated to add the condition inside the module, instead of turning every schedule into a conditional one.

Reply:
I was also thinking about that and it is for sure rather simple solution.

But at the same time it breaks our strategy of not having conditional logic that changes behavior of the test. We are in progress of refactoring our tests to be atomic and to remove extra knowledge about product (i.e. do not have steps that do not belong to the test case that is covered in perl test module). The condition will hide the steps performed for beta popup handling inside of the module, so that we'll not be able to distinguish where we cover that test case without reviewing logs. 

This is one of the things that we try to achieve: make test suites coverage transparent and delegate knowledge of what should be tested to test suite schedule.

I also agree that having the same conditional logic in all the files is not that good from DRY prospective, but this is the drawback of the decision we made some time ago: duplication over complexity in test schedule. We also started discussions on how that could be improved, so that we'll not have so many places to change, once something that is needed for all the test suites is added.

Reply:
> Maybe it would make it less complicated to add the condition inside the module

One more thing, that I forgot to mention, is that the popup does not appear on Product Selection Page. So, technically, it does not belong to Product Select Page and cannot be handled by `select_product` test module.

Reply:
I also 50/50 with that, I think we have a whole new life cycle of the product to figure out better solution, I would just go with adding the test unconditionally, instead of having more mess in every yaml and just create a ticket to work on it.

On the other hand I like that you make the problem more visible and not hide it, so can encourage us to find soon some solution. 
Ideally we should not depend on any openQA settings, so we can just have some notification before it would happen at the end of product life cycle (that is feasible) and enable in single place. It could also helps if some intermediate build doesn't have and we want to continue testing (but I'm not sure if that case make sense...as the build could be invalid). So that is the thing, that you prepare it for not having Beta as well and that might not needed for now.

In summary, I would prefer to have just the test unconditionally, but I'm fine also with this solution.

Reply:
But this particular solution would be very costly if applied in more occasions. I do not propose to merge it with select_product module. I would prefer to keep it separate and the worker would just check a condition at the beginning of the module and do nothing in case of BETA=0. It's not ideal indeed, but keeps the schedules clean.

Reply:
Maybe we can add some extra feature in scheduling for conditions manipulated by vars.

Reply:
> But this particular solution would be very costly if applied in more occasions. I do not propose to merge it with select_product module. I would prefer to keep it separate and the worker would just check a condition at the beginning of the module and do nothing in case of BETA=0. It's not ideal indeed, but keeps the schedules clean.

long history of conditions in modules make me disagree :) I still would prefer this solution (despite some of my comments) to follow with the design that we are currently working on.

Reply:
> Maybe we can add some extra feature in scheduling for conditions manipulated by vars.

that will be done in future, but now cannot be seen the pattern, we need first to have complete installation to do it.

Reply:
One thing is that we also have the GMC flag, which is something of the future, but will also need to happen. In any case, does the merging of yaml schedule suport conditional schedules? if so all that could ble handled within one and then imported as yast installation base or something, from what I see, it's always the same three lines.

Reply:
@foursixnine Yeah, this is one of the solutions we are right now thinking about. We'll address this in another ticket, as right now we don't have some well-established structure for inheritance in `schedule` section of YAML schedule, and we want to think about the proper design first.

Currently this change will affect only YaST Job Group where we'll be able to distinguish all pros and cons of the solution, then we'll be able to apply it for test suites that use installation modules in another Job Groups.

So, in the end, we discussed in the team and agreed that the simplest solution in terms of adapting it in future for new schedule design, will be scheduling of `validate_beta_warning` in all the tests (in YaST job group), as soon as this warning will be with us for a long period of time. Until it will disappear we'll have some solution how to manage it in a convenient way.

Reply:
@OleksandrOrlov awesome, please if you remember... keep me in the loop :D